src/Messages/OWSMessageSender.m

```diff
@@ -607,11 +630,31 @@ - (void)groupSend:(NSArray<SignalRecipient *> *)recipients
     [self saveGroupMessage:message inThread:thread];
     NSMutableArray<TOCFuture *> *futures = [NSMutableArray array];
 
+    NSArray<NSString *> *blockedPhoneNumbers = _blockingManager.blockedPhoneNumbers;
+    int blockedCount = 0;
     for (SignalRecipient *rec in recipients) {
-        // we don't need to send the message to ourselves, but otherwise we send
-        if (![[rec uniqueId] isEqualToString:[TSStorageManager localNumber]]) {
-            [futures addObject:[self sendMessageFuture:message recipient:rec thread:thread]];
+        // We don't need to send the message to ourselves...
+        if ([[rec uniqueId] isEqualToString:[TSStorageManager localNumber]]) {
+            continue;
         }
+        // ...or to anyone on our blocklist...
+        OWSAssert([rec uniqueId].length > 0);
+        if ([blockedPhoneNumbers containsObject:[rec uniqueId]]) {
```

Comment:
I've been tempted myself.  The problem is that `OWSBlockingManager` does synchronization.  Some of the likely users of `[OWSBlockingManager isPhoneNumberBlocked:]` will be perf-sensitive, so I was wary of adding a method that we'd later use not realizing that it was a little expensive.  But yeah, it'd be nice to DRY this up.  I could go either way.